src/components/AnonymousChat/PostNotificationChannelItem.tsx

```diff
@@ -90,15 +76,17 @@ const AnonPostNotificationChannelItem: (props: MessageChannelItemProps) => React
   const isOwnPost = postNotifItem?.rawJson?.isOwnPost;
   const isOwnSignedPost = item?.rawJson?.isOwnSignedPost;
 
-  const type = helperDeterminePostType(isOwnPost, firstComment?.reaction?.isOwningReaction);
+  const {determinePostType} = useChannelHook();
+  const type = determinePostType(postNotifItem);
 
   return (
     <BaseChannelItem
+      type={type}
       anonPostNotificationUserInfo={anonymousPostNotificationUserInfo}
       block={item?.rawJson?.block}
       comments={item?.rawJson?.comments?.length}
       downvote={item?.rawJson?.downvote}
-      isCommentExists={!!firstComment}
+      isCommentExists={Boolean(firstComment)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `helperDeterminePostType` has been replaced with a hook `useChannelHook`, which returns a function `determinePostType`. Ensure that this hook and its returned function are correctly implemented and tested. Also, the prop `isCommentExists` now uses the `Boolean` constructor for conversion, which is more explicit than the previous double negation.

```diff
- const type = helperDeterminePostType(isOwnPost, firstComment?.reaction?.isOwningReaction);
+ const {determinePostType} = useChannelHook();
+ const type = determinePostType(postNotifItem);
- isCommentExists={!!firstComment}
+ isCommentExists={Boolean(firstComment)}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->